src/ca/verify.ts

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2022 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { VerificationError } from '../error';
+import * as sigstore from '../types/sigstore';
+import { x509Certificate } from '../x509/cert';
+import { verifyCertificateChain } from '../x509/verify';
+
+export function verifySigningCertificate(
+  bundle: sigstore.Bundle,
+  trustedRoot: sigstore.TrustedRoot,
+  options: sigstore.ArtifactVerificationOptions_CtlogOptions
+) {
+  if (bundle.verificationMaterial?.content?.$case !== 'x509CertificateChain') {
+    throw new VerificationError('No certificate chain in bundle');
+  }
+
+  const bundleCerts = parseCerts(
+    bundle.verificationMaterial.content.x509CertificateChain.certificates
+  );
+
+  // Check that a trusted certificate chain can be found for the signing
+  // certificate in the bundle
+  const trustedChain = verifyChain(
+    bundleCerts,
+    trustedRoot.certificateAuthorities
+  );
+
+  // Unless disabled, verify the SCTs in the signing certificate
+  if (!options.disable) {
```

Comment:
Should this option be more explicit, like `disableSCTVerification`? Also when might we want to do this?

Reply:
From the Sigstore Client Spec:
> If CT log verification is required for the signing certificate, verify that the certificate has a number of SCTs from the set of trusted CT log keys, larger or equal to the provided threshold.

But not really clear to me why CT log verification would **not** be required.

Reply:
Oh I see, yeah seems sound if this is specced out in the protobufs 👍 

@kommendorkapten do you have any context on why you you might want to disable CT log verification? Like for someone self-hosting Fulcio who hasn't set up SCTs?